components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/util/Utils.java

```diff
@@ -938,13 +939,24 @@ public static String resolveEventName(String notificationChannel) {
     /**
      * Validate email username.
      *
-     * @param username Tenant aware username of the user.
-     * @throws IdentityRecoveryClientException If username is not an email when email username is enabled.
+     * @param user User .
+     * @throws IdentityRecoveryClientException If username is not an email for tenanted users when email username
+     * is enabled.
      */
-    public static void validateEmailUsername(String username) throws IdentityRecoveryClientException {
+    public static void validateEmailUsername(User user) throws IdentityRecoveryClientException {
 
-        if (IdentityUtil.isEmailUsernameEnabled() && StringUtils.countMatches(username, "@") == 0) {
-            throw handleClientException(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_USERNAME, username);
+        if (!IdentityUtil.isEmailUsernameEnabled()) {
+            // Doesn't need to check for the username if "email address as the username" option is disabled.
+            return;
+        }
+        if (MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(user.getTenantDomain())) {
+            // Super tenant user should be able to log in with both email username or non-email username.
+            return;
+        }
+        if (StringUtils.countMatches(user.getUserName(), "@") == 0) {
```

Comment:
use constant for "@"

Reply:
Fixed with https://github.com/wso2-extensions/identity-governance/pull/596